pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -957,6 +957,7 @@ type PrometheusSpec struct {
 	RetentionSize ByteSize `json:"retentionSize,omitempty"`
 
 	// When true, the Prometheus compaction is disabled.
+	// Defaults to true if `spec.thanos.objectStorageConfig` or `spec.objectStorageConfigFile` are defined.
```

Comment:
(suggestion) let's explain the rationale.
```suggestion
	// When `spec.thanos.objectStorageConfig` or `spec.objectStorageConfigFile` are defined, the operator automatically disables block compaction to avoid race conditions during block uploads (as the Thanos documentation recommends).
```